numpy/core/tests/test_errstate.py

```diff
@@ -39,3 +40,14 @@ def foo(*args):
             with np.errstate(call=None):
                 assert_(np.geterrcall() is None, 'call is not None')
         assert_(np.geterrcall() is olderrcall, 'call is not olderrcall')
+
+    def test_errstate_decorator(self):
+        @np.errstate(all='ignore')
+        def foo():
+            a = -np.arange(3)
+            a // 0
+            
+        foo()
+
+if __name__ == "__main__":
```

Comment:
If this addition still needed?

Reply:
🤷‍♂️ https://github.com/numpy/numpy/blob/b047f387d2424bd2134db6005190311b15a84ac2/doc/TESTS.rst.txt

Says it's OK, but the notes for 1.14.0 says it'll break. Might wanna sort that out.

Reply:
Not wanted, and wouldn't work anyway without importing `run_module_suite`, which only works for Nose.

Reply:
The documentation needs a fix, @mattip ping.

Reply:
removed.

Reply:
@charris #12787